Carbon.IIS/Functions/Set-CIisConfigurationAttribute.ps1

```diff
@@ -0,0 +1,194 @@
+        # You can use a hashtable to also set multiple attributes at once by piping the hashtable's enumerator to
+        # `Get-CIisConfigurationSection`, e.g. `@{}.GetEnumerator() | Set-CIisConfigurationAttribute`.
+        [Parameter(Mandatory, ValueFromPipelineByPropertyName)]
+        [Alias('Key')]
+        [String] $Name,
+
+        # The attribute's value.
+        #
+        # You can pipe objects with `Name` and `Value` properties to `Get-CIisConfigurationAttribute` to set multiple
+        # attribute values at once.
+        #
+        # You can use a hashtable to also set multiple attributes at once by piping the hashtable's enumerator to
+        # `Get-CIisConfigurationSection`, e.g. `@{}.GetEnumerator() | Set-CIisConfigurationAttribute`.
+        [Parameter(Mandatory, ValueFromPipelineByPropertyName)]
+        [AllowNull()]
+        [AllowEmptyString()]
+        [Object] $Value,
+
+        # If the attribute's value is sensitive. If set, the attribute's value will be masked when written to the
+        # console.
+        [bool] $Sensitive
+    )
+
+    begin
+    {
+        Set-StrictMode -Version 'Latest'
+        Use-CallerPreference -Cmdlet $PSCmdlet -Session $ExecutionContext.SessionState
+        $section = Get-CIisConfigurationSection -SiteName $SiteName -VirtualPath $VirtualPath -SectionPath $SectionPath
+        if( -not $section )
+        {
+            return
+        }
+
+        $attrNameFieldLength =
+            $section.Attributes |
+            Select-Object -ExpandProperty 'Name' |
+            Select-Object -ExpandProperty 'Length' |
+            Measure-Object -Maximum |
+            Select-Object -ExpandProperty 'Maximum'
+        $nameFormat = "{0,-$($attrNameFieldLength)}"
+
+        $attrNames = [Collections.ArrayList]::New()
+        $commitChanges = $false
+
+        $basePrefix = "[IIS:/Sites/$(Join-CIisVirtualPath -Path $SiteName -ChildPath $VirtualPath):$($SectionPath)"
+    }
+
+    process
+    {
+        if( -not $section )
+        {
+            return
+        }
+
+        $Name = "$($Name.Substring(0, 1).ToLowerInvariant())$($Name.Substring(1, $Name.Length -1))"
+
+        $msgPrefix = "$($basePrefix)@$($nameFormat -f $Name)]  "
+
+        $currentValue = $section.GetAttributeValue($Name)
+
+        $protectValue = $Sensitive
+        if( $Value -is [SecureString] )
+        {
+            $Value = [pscredential]::New('i', $Value).GetNetworkCredential().Password
+            $protectValue = $true
+        }
+        elseif( $Value -is [switch] )
+        {
+            $Value = $Value.IsPresent
+        }
+
+        $currentValueMsg = $currentValue
+        $valueMsg = $Value
+
+        if( $Value -is [Enum] )
+        {
+            $currentValueMsg = [enum]::GetName($Value.GetType().FullName, $currentValue)
+        }
+
+        if( $protectValue )
+        {
+            $currentValueMsg = '*' * $currentValueMsg.ToString().Length
+            $valueMsg = '*' * $valueMsg.ToString().Length

```

Comment:
Always mask with a static length for extra security?

Reply:
Yes.